packages/app/universal-ts-utils/src/array/uniq.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { describe, expect, it } from 'vitest'
+import { uniq } from './uniq'
+
+describe('uniq', () => {
+  it('returns a new array of mixed primitive value without duplicates', () => {
+    const objectA = {}
+    const objectB = {}
+    const duplicateValues = [
+      1,
+      1,
+      'a',
+      'a',
+      Number.NaN,
+      Number.NaN,
+      true,
+      true,
+      false,
+      false,
+      null,
+      null,
+      undefined,
+      undefined,
+      objectA,
+      objectA,
+      objectB,
+      objectB,
+    ]
+
+    expect(uniq(duplicateValues)).toEqual([
+      1,
+      'a',
+      Number.NaN,
+      true,
+      false,
+      null,
+      undefined,
+      objectA,
+      objectB,
```

Comment:
🟡 Is this intended to work with objects? for different (in memory) objects but with the same content this method won't work, wondering if it can be confusing. 
If we are fine with it, all good, if not I would say we can try to specify which types are accepted by the method